packages/validator/src/rules/no-undefined-path-parameters/no-undefined-path-parameters.ts

```diff
@@ -0,0 +1,56 @@
+import { constructHttpPath } from "@fern-api/ir-generator";
+import chalk from "chalk";
+import { Rule, RuleViolation } from "../../Rule";
+
+export const NoUndefinedPathParametersRule: Rule = {
+    name: "no-duplicate-enum-values",
+    create: () => {
+        return {
+            httpEndpoint: ({ endpoint }) => {
+                const errors: RuleViolation[] = [];
+
+                const urlPathParameters = new Set();
+                if (endpoint.path != null) {
+                    const httpPath = constructHttpPath(endpoint.path);
+                    httpPath.parts.forEach((part) => {
+                        if (urlPathParameters.has(part.pathParameter)) {
+                            errors.push({
+                                severity: "error",
+                                message: `Endpoint path has duplicate path parameter: ${chalk.bold(
+                                    part.pathParameter
+                                )}`,
+                            });
+                        }
+                        urlPathParameters.add(part.pathParameter);
+                    });
+                }
+
+                const definedPathParameters = new Set(Object.keys(endpoint["path-parameters"] ?? {}));
```

Comment:
nbd, but the only thing about using sets everywhere is that you lose the order of how things appear in the yaml.

Reply:
gotcha so within the same endpoint things can get jumbled?